test/adoption_center/test_client.rb

```diff
@@ -0,0 +1,18 @@
+require 'minitest/autorun'
+require '../lib/client'
+include Client
```

Comment:
Great start!

```diff
+require '../lib/client'
+include Client
```

I don't think we should use the relative require of lib/client, but should just have `require 'test_helper'`

I think including the Client into the main scope here is problematic for a couple of reasons:

1. that's not how it will be used
2. mixing modules into the main scope often causes test interaction weirdness
3. It makes it look like `Client.authenticate` as called in the tests is part of the Client api, but really `authenticate` it's just a globally available method.  What you probably mean here was `class TestClient; include Client; end` and `client = TestClient.new` and `client.authenticate(email, password)`.

Specifically here, for a client that is stateful, I think it's better to use a class to encapsulate logging in, etc.

e.g. 

```ruby
class RubygemsOrgClient
  def initialize
    self.authenticated = false
  end
  def authenticate(email, password)
    # login
    self.authenticated = true
  end
  def authenticated?
   authenticated === true
  end

  private
  attr_accessor :authenticated
end
```

Used as

```ruby
client = RubygemsOrgClient.new
```

Also, based on the code above, looks like we could save some time by wrapping the [`gems` gem](https://github.com/rubygems/gems). 

```ruby
require 'rubygems'
require 'gems'

# Retrieve your API key using HTTP basic authentication.
Gems.configure do |config|
  config.username = 'nick@gemcutter.org'
  config.password = 'schwwwwing'
end
Gems.api_key

# The following methods require authentication.
# By default, we load your API key from ~/.gem/credentials
# You can override this default by specifying a custom API key.
Gems.configure do |config|
  config.key = '701243f217cdf23b1370c7b66b65ca97'
end


# Return all gems that you own.
Gems.gems
```